packages/react-dom/src/__tests__/ReactDOMEventPropagation-test.js

```diff
@@ -0,0 +1,1327 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+'use strict';
+
+describe('ReactDOMEventListener', () => {
+  let React;
+  let OuterReactDOM;
+  let InnerReactDOM;
+  let container;
+
+  beforeEach(() => {
+    jest.resetModules();
+    React = require('react');
+    jest.isolateModules(() => {
+      OuterReactDOM = require('react-dom');
+    });
+    jest.isolateModules(() => {
+      InnerReactDOM = require('react-dom');
+    });
+    expect(OuterReactDOM).not.toBe(InnerReactDOM);
+  });
+
+  afterEach(() => {
+    cleanup();
+  });
+
+  function cleanup() {
+    if (container) {
+      OuterReactDOM.unmountComponentAtNode(container);
+      document.body.removeChild(container);
+      container = null;
+    }
+  }
+
+  function render(tree) {
+    cleanup();
+    container = document.createElement('div');
+    document.body.appendChild(container);
+    OuterReactDOM.render(tree, container);
+  }
+
+  describe('bubbling events', () => {
+    // This test will fail in legacy mode (only used in WWW)
+    // because we emulate the React 16 behavior where
+    // the click handler is attached to the document.
+    // @gate !enableLegacyFBSupport
+    it('onClick', () => {
+      testNativeBubblingEvent({
+        type: 'div',
+        reactEvent: 'onClick',
+        nativeEvent: 'click',
+        dispatch(node) {
+          node.click();
+        },
+      });
+    });
+
+    it('onKeyPress', () => {
+      testNativeBubblingEvent({
```

Comment:
However, the actual tests are a sequence of several in each case. I didn't want to repeat because we want consistent behavior across events in the same group. So we call a helper function that runs all tests. If you want to run one case in particular you can always comment out the calls inside of that function.